property/collector.go

```diff
@@ -96,7 +124,27 @@ func (p *Collector) CreateFilter(ctx context.Context, req types.CreateFilter) er
 	return nil
 }
 
-func (p *Collector) WaitForUpdates(ctx context.Context, version string, opts ...*types.WaitOptions) (*types.UpdateSet, error) {
+func (p *Collector) CreateFilterEx(ctx context.Context, req types.CreateFilter) (*Filter, error) {
+	req.This = p.Reference()
+
+	resp, err := methods.CreateFilter(ctx, p.roundTripper, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Filter{roundTripper: p.roundTripper, reference: resp.Returnval}, nil
+}
+
+func (p *Collector) WaitForUpdates(
+	ctx context.Context,
+	version string,
+	opts ...*types.WaitOptions) (*types.UpdateSet, error) {
+
+	if err := p.lock(ctx); err != nil {
```

Comment:
Rather than block, should we return an error or even panic if a Collector instance is already in WaitForUpdates? And document a Collector instance's WaitForUpdates can only be used by a single thread?

Reply:
Will do. I'll switch to using `TryLock` from a mutex and if it fails, either panic or return an error.